app/services/scene-collections/state.ts

```diff
@@ -245,4 +264,46 @@ export class SceneCollectionsStateService extends StatefulService<ISceneCollecti
       Vue.set(this.state, key, state[key]);
     });
   }
+
+  @mutation()
+  CREATE_NODE_MAP_ENTRY(sceneId: string, horizontalNodeId: string, verticalNodeId: string) {
+    const activeId = this.state.activeId;
+    const coll = this.state.collections.find(coll => coll.id === activeId);
+    // confirm or set node map
+    if (!coll) return;
+    if (!coll.sceneNodeMaps) coll.sceneNodeMaps = {};
+    if (!coll.sceneNodeMaps[sceneId]) coll.sceneNodeMaps[sceneId] = {};
+
+    coll.sceneNodeMaps[sceneId] = {
+      ...coll.sceneNodeMaps[sceneId],
+      [horizontalNodeId]: verticalNodeId,
+    };
+  }
+
+  @mutation()
+  REMOVE_VERTICAL_NODE(horizontalNodeId: string, sceneId: string) {
+    const activeId = this.state.activeId;
+    const coll = this.state.collections.find(coll => coll.id === activeId);
+
+    // confirm existence of scene node map
+    if (!coll || !coll.sceneNodeMaps || !coll.sceneNodeMaps[sceneId]) return;
+
+    const nodeMap = coll.sceneNodeMaps[sceneId];
+    delete nodeMap[horizontalNodeId];
```

Comment:
why is this called `REMOVE_VERTICAL_NODE` but it's deleting a horizontal node?

Reply:
In the node map, the horizontal node id is the key for the vertical node id

Reply:
that makes sense, but is a little counter-intuitive i think, might want to add an explanatory comment